File: src/Client/src/models/dtos/update-profile-dto.ts
import { Gender } from "../enums/gender";

export class UpdateProfileDto {
    public username : string = '';
    public firstName : string = '';
    public lastName :string = '';
    public phoneNumber : string = '';
    public birthdate : string = ''
    public biography : string = '';
    public gender : Gender = Gender.other;
}

File: src/Api/Models/DTOs/UpdateProfileDTO.cs
using Api.Enums;

namespace Api.DTOs;
public class UpdateProfileDTO
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string? Biography { get; set; }
    public string? Username { get; set; }
    public DateTime Birthdate { get; set; } = new();
    public Gender Gender { get; set; }
}